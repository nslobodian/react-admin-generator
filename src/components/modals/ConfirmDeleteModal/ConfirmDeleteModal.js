import React from 'react'
import connectModal from 'redux-modal/lib/connectModal'
import Modal from 'react-bootstrap/lib/Modal'
import withHandlers from 'recompose/withHandlers'

export const ConfirmDeleteModal = withHandlers({
  onDeleteHandler: ({ handleHide, handleDelete }) => () => {
    handleDelete()
    handleHide()
  },
})(({ show, handleHide, onDeleteHandler, item, deleteMessage }) =>
  <Modal
    show={show}
    onHide={handleHide}
  >
    <Modal.Header closeButton>
      <Modal.Title>Delete {item.toUpperCase()}</Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <div>
        { deleteMessage || `Are you sure you want to delete this awesome ${item}?` }
      </div>
    </Modal.Body>
    <Modal.Footer>
      <button className='btn btn-default' onClick={handleHide}>Close</button>
      <button className='btn btn-danger' onClick={onDeleteHandler}>Delete</button>
    </Modal.Footer>
  </Modal>
  )

export default connectModal({ name: 'confirmDeleteModal', destroyOnHide: true })(ConfirmDeleteModal)
