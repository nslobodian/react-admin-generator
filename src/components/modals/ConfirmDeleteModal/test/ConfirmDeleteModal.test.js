import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ConfirmDeleteModal } from '../ConfirmDeleteModal'

const props = {
  handleHide: jest.fn(),
  handleDelete: jest.fn(),
  show: true,
  item: 'item',
}

describe('(Components) ConfirmDeleteModal', () => {
  it('Should render component properly', () => {
    const wrapper = shallow(
      <ConfirmDeleteModal
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
