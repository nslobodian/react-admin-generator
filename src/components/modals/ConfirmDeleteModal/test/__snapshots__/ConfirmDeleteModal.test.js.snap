// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) ConfirmDeleteModal Should render component properly 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Delete 
      ITEM
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      Are you sure you want to delete this awesome 
      item
      ?
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-default"
      onClick={[Function]}
    >
      Close
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete
    </button>
  </ModalFooter>
</Modal>
`;
